### PR TITLE
feat(python): infer timezone and improve display

### PR DESCRIPTION
**Pull request by ritchie46**
This improves timezone aware display. The display is now exactly similar to the timezone `datetime` objects you get in python. This was a source of confusion and bugs because we used pyarrows display, which shows the UTC timestamp no matter what the timezone was.

Next we also set the timezone automatically if the first non null `datetime` value contains a timezone.

### CODE REVIEW

**Comment by ritchie46** [comment]
@alexander-beedie the `init_timezone` tests succeeds locally! :tada: 

I think I fixed it in #5226

File: py-polars/polars/utils.py [hunk 1]
```diff
@@ -197,6 +197,7 @@ def _prepare_row_count_args(
 
 
 EPOCH = datetime(1970, 1, 1).replace(tzinfo=None)
+UTC = zoneinfo.ZoneInfo("UTC")
```
Comment (jaychia) [inline py-polars/polars/utils.py:200]: FYI @ritchie46 - this breaks imports of Polars in Google Colab (Python 3.7) if installed as `polars` without `polars[timezone]`:

```
!pip install polars

import polars
```

```
[/usr/local/lib/python3.7/dist-packages/polars/utils.py](https://localhost:8080/#) in <module>
    198 
    199 EPOCH = datetime(1970, 1, 1).replace(tzinfo=None)
--> 200 UTC = zoneinfo.ZoneInfo("UTC")
    201 
    202 

NameError: name 'zoneinfo' is not defined
```

Is it possible to handle the `_ZONEINFO_AVAILABLE=False` case a little more elegantly, and to specify `backports.timezone` as a dependency if the Python version is <3.9?

Comment (dhirschfeld) [reply py-polars/polars/utils.py:200]: ***xref:*** #5244

Comment (ritchie46) [reply py-polars/polars/utils.py:200]: Ai.. Will patch this today

**Pull request merged**